syntax = "proto3";

package creative_campaign;

// CreativeGenerateRequest is sent when a creative needs to be generated
message CreativeGenerateRequest {
  string campaign_id = 1;
  string locale = 2;
  string status = 3;  // e.g., "pending", "generated", "failed"
  string error_message = 4;  // Only set if status is "failed"
  
  // Timestamps
  string created_at = 10;
  string updated_at = 11;
}

// CreativeGenerated is published when a creative has been generated
message CreativeGenerated {
  string campaign_id = 1;
  string locale = 2;
  
  // Generated content
  string headline = 3;
  string description = 4;
  string call_to_action = 5;
  repeated string visual_elements = 6;  // Descriptions of visual elements
  
  // Status
  string status = 7;  // "generated" or "failed"
  string error_message = 8;  // Only set if status is "failed"
  
  // Timestamps
  string generated_at = 20;
}
